tests/unit/configFile.spec.js

```diff
@@ -15,69 +18,96 @@ describe("configFile.js", () => {
-    var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
-    settings.configEncrypted = false;
-    await LocalStore.setOne(StoreKey.SETTINGS, settings);
-    fetch.mockResponseOnce(encryptedSettings.encryptedData);
+    it("errorMsg shouldn't be null if file is encrypted but encryption is disabled", async () => {
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      settings.configEncrypted = false;
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      fetch.mockResponseOnce(encryptedSettings.encryptedData);
 
-    const result = await updateNow();
-    var lastConfigData = await LocalStore.getOne(StoreKey.LAST_CONFIG_DATA);
-    expect(lastConfigData.errorMsg).not.toBe(null);
-    expect(result).toBe(false);
+      const result = await updateNow();
+      var lastConfigData = await LocalStore.getOne(StoreKey.LAST_CONFIG_DATA);
+      expect(lastConfigData.errorMsg).not.toBe(null);
+      expect(result).toBe(false);
+    });
+
+    it("errorMsg shouldn't be null if file is encrypted but encryption key is wrong", async () => {
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      settings.configEncrypted = true;
+      settings.configEncryptionKey = "wrong password";
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      fetch.mockResponseOnce(encryptedSettings.encryptedData);
+
+      const result = await updateNow();
+      var lastConfigData = await LocalStore.getOne(StoreKey.LAST_CONFIG_DATA);
+      expect(lastConfigData.errorMsg).not.toBe(null);
+      expect(result).toBe(false);
+    });
   });
 
-  it("errorMsg shouldn't be null if file is encrypted but encryption key is wrong", async () => {
-    var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
-    settings.configEncrypted = true;
-    settings.configEncryptionKey = "wrong password";
-    await LocalStore.setOne(StoreKey.SETTINGS, settings);
-    fetch.mockResponseOnce(encryptedSettings.encryptedData);
-
-    const result = await updateNow();
-    var lastConfigData = await LocalStore.getOne(StoreKey.LAST_CONFIG_DATA);
-    expect(lastConfigData.errorMsg).not.toBe(null);
-    expect(result).toBe(false);
+  describe("generateJSONFile function", () => {
+    it("generateJSONFile result should be equal to simple configuration file", async () => {
+      await LocalStore.clear();
+      await parseJSONFile(defaultSettings, true);
+
+      const result = await generateJSONFile();
+      expect(result).toStrictEqual(_.omit(defaultSettings, "update"));
+    });
+
+    it("generateJSONFile result should be equal to changed configuration file", async () => {
+      await LocalStore.clear();
+      await parseJSONFile(defaultSettings, true);
+
+      // Change data in local storage.
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      settings.configEncrypted = true;
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
```

Comment:
Please change more things... 

* Change two or three values from settings.
* Change the name of one group.
* Remove a search provider, add a new search provider and modify the name of an existing search provider.
* Change a special provider (modify two values from Config and on Queries remove a value and update an existing one).